src/app/clusters/bindings/BindingManager.cpp

```diff
@@ -101,31 +101,28 @@ CHIP_ERROR BindingManager::EstablishConnection(const ScopedNodeId & nodeId)
     VerifyOrReturnError(mInitParams.mCASESessionManager != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
     mLastSessionEstablishmentError = CHIP_NO_ERROR;
-    mInitParams.mCASESessionManager->FindOrEstablishSession(nodeId, &mOnConnectedCallback, &mOnConnectionFailureCallback);
+    auto * connectionCallback      = Platform::New<ConnectionCallback>(this);
+    mInitParams.mCASESessionManager->FindOrEstablishSession(nodeId, connectionCallback->GetOnDeviceConnected(),
+                                                            connectionCallback->GetOnDeviceConnectionFailure());
     if (mLastSessionEstablishmentError == CHIP_ERROR_NO_MEMORY)
```

Comment:
Does this actually work the way it's intended to? I don't believe `CHIP_ERROR_NO_MEMORY` is actually emitted when there are no more entries in the `mPendingNotificationMap`....

I know it isn't germane to this PR, but if it's a bug, let's file a follow-up.

Reply:
This error would not come from mPendingNotificationMap it would come from CASESessionManager. This is the issues I was trying to communicate in https://github.com/project-chip/connectedhomeip/pull/22133#discussion_r955432949

Regardless this is something incorrect that need to be fixed in a follow up PR and arguably after post 1.0
https://github.com/project-chip/connectedhomeip/issues/22173